src/format/parse.rs

```diff
@@ -224,6 +225,263 @@ fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseResult<(&'a st
+            fraction_to_seconds(fraction, min(digits, MAX_FRACTION_DIGITS), 3600);
+
+        parsed.set_minute(seconds / 60)?;
+        parsed.set_second(seconds % 60)?;
+        parsed.set_nanosecond(nanos)?;
+        return Ok((s, format, hour == 24));
+    }
+
+    let c = s.as_bytes().first().unwrap_or(&b'a');
+    if !(c.is_ascii_digit() || c == &b':') {
+        // Allow reduced accuracy
+        parsed.set_minute(0)?;
+        return Ok((s, format, hour == 24));
+    }
+
+    format = if s.as_bytes().first() == Some(&b':') { Extended } else { Basic };
+    if format == Extended {
+        s = &s[1..];
+    }
+    parsed.set_minute(try_consume!(scan::number(s, 2, 2)))?;
+
+    if let Some(&b',' | &b'.') = s.as_bytes().first() {
+        // Second and nanosecond are expressed as a fraction of a minute.
+        s = &s[1..];
+        let digits = s.as_bytes().iter().take_while(|c| c.is_ascii_digit()).count();
+        let (_, fraction) = scan::number(s, 1, MAX_FRACTION_DIGITS)?;
+        s = &s[digits..];
+        let (seconds, nanos) = fraction_to_seconds(fraction, min(digits, MAX_FRACTION_DIGITS), 60);
+
+        parsed.set_second(seconds)?;
+        parsed.set_nanosecond(nanos)?;
+        return Ok((s, format, hour == 24));
+    }
+
+    let c = s.as_bytes().first().unwrap_or(&b'a');
+    if !(c.is_ascii_digit() || (format == Extended && c == &b':')) {
+        // Allow reduced accuracy
+        return Ok((s, format, hour == 24));
+    }
+
+    if format == Extended {
+        s = scan::char(s, b':')?;
+    }
+    parsed.set_second(try_consume!(scan::number(s, 2, 2)))?;
+
+    if let Some(&b',' | &b'.') = s.as_bytes().first() {
+        let nano = try_consume!(scan::nanosecond(&s[1..]));
+        parsed.set_nanosecond(nano)?;
+    }
+
+    if hour == 24
+        && (parsed.minute.unwrap_or(0) != 0
+            || parsed.second.unwrap_or(0) != 0
+            || parsed.nanosecond.unwrap_or(0) != 0)
+    {
+        return Err(INVALID);
+    }
+    Ok((s, format, hour == 24))
+}
+
+fn fraction_to_seconds(fraction: i64, digits: usize, max_seconds: i64) -> (i64, i64) {
+    const SCALE: [i64; 15] = [
+        1, // unused, for easy indexing
+        100_000_000_000_000,
+        10_000_000_000_000,
+        1_000_000_000_000,
+        100_000_000_000,
+        10_000_000_000,
+        1_000_000_000,
+        100_000_000,
+        10_000_000,
+        1_000_000,
+        100_000,
+        10_000,
+        1_000,
+        100,
+        10,
+    ];
+
+    let scale = SCALE[digits];
+    let huge = fraction * max_seconds * scale;
+    let seconds = huge / 1_000_000_000_000_000;
+    let nanos = (huge % 1_000_000_000_000_000) / 1_000_000;
+    (seconds, nanos)
+}
+
+// This method returns a `NaiveDateTime` instead of working with `Parsed` because `Parsed` can't
+// handle a time of `24:00:00` (which should parse to `00:00:00` the next day).
+pub(crate) fn parse_iso8601_datetime(s: &str) -> ParseResult<(NaiveDateTime, &str, Iso8601Format)> {
```

Comment:
Please describe the meaning of returned `&str` value.

Reply:
This should describe the `remainder`.

    - `remainder` is the remaining part of the parsed string `s`.